static/js/profile.js

```diff
@@ -0,0 +1,539 @@
+
+*/
+	var profile = {};
+	var apisecret = "", storeapisecret = false;
+	
+	var defaultprofile = {
+			//General values
+			"simple": true,
+			"calculator": false,
+			"dia":3,
+			
+			// Simple style values, "from" are in minutes from midnight
+			"carbratio":24,
+			"carbs_hr":30,
+			"sens":17,
+			
+			"validfrom": new Date(),
+			
+			//Bolus calculator style values, "from" are in minutes from midnight
+			"ic": [{"from":0,"val":24},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0}
+					],
+			"car_high": 30,
+			"car_medium": 30,
+			"car_low": 30,
+			"isf": [{"from":0,"val":17},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0}
+					],
+
+			// Basal rates
+			"basal": [{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0},
+					{"from":0,"val":0}
+					]
+	};
+
+	var GuiToVal = [
+			// API secret
+			{ "html":"pe_apisecret", 		"type":"text" , 	"settings":"apisecret" },
+			{ "html":"pe_storeapisecret",	"type":"checkbox" , "settings":"storeapisecret" },
+	
+			// General
+			{ "html":"pe_input_simple",		"type":"checkbox" , "settings":"profile.simple" },
+			{ "html":"pe_input_calculator",	"type":"checkbox" , "settings":"profile.calculator" },
+			{ "html":"pe_dia", 				"type":"float" , 	"settings":"profile.dia" },
+			
+			{ "html":"pe_validfrom", 		"type":"date" , 	"settings":"profile.validfrom" },
+			
+			// Simple style values
+			{ "html":"pe_if", 			"type":"int" , 			"settings":"profile.carbratio" },
+			{ "html":"pe_hr", 			"type":"int" , 			"settings":"profile.carbs_hr" },
+			{ "html":"pe_isf", 			"type":"int" , 			"settings":"profile.sens" },
+			
+			//Bolus calculator style values
+			{ "html":"pe_hr_high", 		"type":"int" , 			"settings":"profile.car_high" },
+			{ "html":"pe_hr_medium",	"type":"int" , 			"settings":"profile.car_medium" },
+			{ "html":"pe_hr_low", 		"type":"int" , 			"settings":"profile.car_low" },
+			
+			{ "html":"pe_if_from_0", 	"type":"dropdownval" ,	"settings":"profile.ic[0].from" },
```

Comment:
Is there a way to create these programatically? I can see how hardcoding the entries was the easiest way to get going, but this would be way cleaner if the number of entries wasn't hardcoded.